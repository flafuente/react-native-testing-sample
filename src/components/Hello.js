import React from 'react';

import {
    View,
    Text,
    TextInput,
    TouchableOpacity,
    Platform,
    StyleSheet
} from 'react-native';

import {
    Actions,
} from 'react-native-router-flux';

    class Hello extends React.Component {
    constructor(props) {
            super(props);
    }
    render() {
        return (
            <View>
                <Text style={styles.title}>
                    {this._composePhrase(this.props.alias)}
                </Text>
                <TouchableOpacity
                    onPress={() => {
                        // Navigate
                        Actions.home({});
                    }}>
                    <Text style={styles.backBtn}>Back</Text>
                </TouchableOpacity>
            </View>
        );
    }
    _composePhrase(alias) {
        return "Hello " + alias;
    }
}

export const styles = StyleSheet.create({
    title: {
        marginTop: 20,
        marginLeft: 20,
        fontSize: 17
    },
    backBtn: {
        marginLeft: 20,
        fontSize: 20
    }
});

export default Hello