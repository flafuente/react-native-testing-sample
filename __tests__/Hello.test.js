import 'react-native';
import React from 'react';
import Hello, { styles } from '../src/components/Hello';
import { render, shallow } from 'enzyme';

import {
  Text
} from 'react-native';

const wrapper = shallow(
  <Hello alias="fernando" />
);

it('Hello component renders correctly', () => {
  // To update snapshot. => npm test -- -u Hello.test
  expect(wrapper).toMatchSnapshot();
});

it('Property alias should contain a Text component with "Hello {alias}"', () => {
  expect(wrapper.contains(
    <Text style={styles.title}>
      Hello fernando
    </Text>
  )).toBe(true);
});