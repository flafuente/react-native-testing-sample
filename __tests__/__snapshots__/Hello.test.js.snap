// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 17,
        "marginLeft": 20,
        "marginTop": 20,
      }
    }
  >
    Hello fernando
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginLeft": 20,
        }
      }
    >
      Back
    </Text>
  </TouchableOpacity>
</View>
`;
